dialer/bandit.go

```diff
@@ -114,6 +114,21 @@ func (o *BanditDialer) chooseDialerForDomain(network, addr string) (ProxyDialer,
 	notAllFailing := hasNotFailing(o.dialers)
 	for i := 0; i < (len(o.dialers) * 2); i++ {
 		d = o.dialers[chosenArm]
+		readyChan := d.Ready()
+		if readyChan != nil {
+			select {
+			case err := <-readyChan:
```

Comment:
Hey sorry, I just realized: this only works once, right?  Once someone reads from the ready channel, there will never again be a value on it.

I think the ready channel should probably only be used to signal readiness and should just be closed when the dialer is ready.  Errors in initialization can be returned by the dial function.

Sorry I didn't catch this earlier!

Reply:
> I think the ready channel should probably only be used to signal readiness and should just be closed when the dialer is ready.

Makes sense to me!

> Errors in initialization can be returned by the dial function.

If we do this, the proxy will be punished by the bandit algorithm, should we still do it?

Reply:
I'll try to create a logic for sending a response for all listeners

Reply:
Ok, I've updated the ready logic from the water implementation.

Whenever we receive a `ready()` call for checking if the dialer is ready:

1. We check if the dialer is ready, if it is we return a nil response
2. We check if there was an error loading the WASM file.
  2.1 In this case, all other ready calls have already received an answer and they're closed, so this is a late party check
  2.2 In this case we return a temp channel and start a go func that send the error and close the channel.
3. We create a channel, append to a internal list of channels and return to whoever is calling `ready()`.

When we finish loading the WASM by either succeeding or failing, we broadcast the ready status to all channels. If the channel take more than one second for receiving the message, the channel is probably not being listened so it'll be closed.

Reply:
> > Errors in initialization can be returned by the dial function.
>
> If we do this, the proxy will be punished by the bandit algorithm, should we still do it?

If there was an error in initialization then the proxy is dead in the water.  In that case, we don't care if it gets punished by the bandit algorithm since it will error every time anyway (actually, we want it to get de-prioritized).

So a simply closed channel, with errors returned on dial attempts would work.  What you've got works too though!  Totally up to you.